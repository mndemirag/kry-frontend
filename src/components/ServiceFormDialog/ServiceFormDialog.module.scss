@import "../../utils";

.overlay {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background: rgba(0, 0, 0, 0.7);
  z-index: 1000;
}

.popup {
  margin: 70px auto;
  padding: 20px;
  background: #fff;
  border-radius: 5px;
  width: 70%;
  position: relative;

  @media (min-width: $screen-md-min) {
    width: 50%;
  }

  h2 {
    margin-top: 0;
    color: #333;
    font-family: Tahoma, Arial, sans-serif;
  }

  .close {
    position: absolute;
    top: 20px;
    right: 30px;
    font-size: 30px;
    font-weight: bold;
    text-decoration: none;
    color: #333;

    :hover {
      color: #06d85f;
    }
  }

  .content {
    max-height: 30%;
  }
}

.serviceForm {
  padding: 0px;

  @media (min-width: $screen-md-min) {
    padding: 30px 30px 12px;
  }
}

.buttonContainer {
  padding-top: 32px;
  text-align: center;
}
