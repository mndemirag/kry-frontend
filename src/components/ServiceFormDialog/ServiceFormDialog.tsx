import { useState } from "react";
import { ServiceFormDialogProps } from "../../types";
import { Button } from "../Button/Button";
import { FormInput } from "../FormInput/FormInput";

import styles from "./ServiceFormDialog.module.scss";

export const ServiceFormDialog: React.FC<ServiceFormDialogProps> = ({
  onClose,
  onSubmit,
}) => {
  const [name, setName] = useState("");
  const [url, setUrl] = useState("");
  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    onSubmit(name, url);
    onClose();
  };

  return (
    <div className={styles.overlay}>
      <div className={styles.popup}>
        <h2>Add a new service</h2>
        <span
          className={styles.close}
          aria-label="Close the form"
          onClick={() => onClose()}
        >
          &times;
        </span>
        <div className={styles.content}>
          <form className={styles.serviceForm} onSubmit={handleSubmit}>
            <FormInput label="Name" onChange={setName} />
            <FormInput label="URL" onChange={setUrl} />

            <div className={styles.buttonContainer}>
              <Button aria-label="Close the form" onClick={onClose}>
                Close
              </Button>
              <Button aria-label="Submit the form" type="submit">
                Submit
              </Button>
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};
