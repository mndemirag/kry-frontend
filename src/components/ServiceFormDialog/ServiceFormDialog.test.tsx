import { render } from "@testing-library/react";

import { ServiceFormDialog } from "./ServiceFormDialog";

const mockClose = jest.fn();
const mockSubmit = jest.fn();

describe("ServiceFormDialog", () => {
  it("renders properly", () => {
    const { container } = render(
      <ServiceFormDialog onClose={mockClose} onSubmit={mockSubmit} />
    );
    expect(container).toMatchSnapshot();
  });
});
