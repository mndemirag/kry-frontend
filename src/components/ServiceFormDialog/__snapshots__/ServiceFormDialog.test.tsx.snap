// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceFormDialog renders properly 1`] = `
<div>
  <div
    class="overlay"
  >
    <div
      class="popup"
    >
      <h2>
        Add a new service
      </h2>
      <span
        aria-label="Close the form"
        class="close"
      >
        Ã—
      </span>
      <div
        class="content"
      >
        <form
          class="serviceForm"
        >
          <label
            class="inputLabel"
          >
            Name
            <input
              aria-label="Name"
              class="inputField"
              type="text"
            />
          </label>
          <label
            class="inputLabel"
          >
            URL
            <input
              aria-label="URL"
              class="inputField"
              type="text"
            />
          </label>
          <div
            class="buttonContainer"
          >
            <button
              aria-label="Close"
              class="button"
            >
              Close
            </button>
            <button
              aria-label="Submit"
              class="button"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
